### PR TITLE
Fix #10785: work around matplotlib warning caused by PyInstaller

### LINKED ISSUE #10785: TST: pyinstaller cron job failed with MATPLOTLIBDATA environment variable was deprecated
Example log: https://travis-ci.org/github/astropy/astropy/jobs/731532639

```
ImportError while loading conftest '/home/travis/build/astropy/astropy/.pyinstaller/astropy_tests/conftest.py'.
_pytest/config/__init__.py:570: in _importconftest
    ???
_pytest/pathlib.py:517: in import_path
    ???
_pytest/assertion/rewrite.py:171: in exec_module
    ???
astropy_tests/conftest.py:22: in <module>
    import matplotlib
../.tox/pyinstaller/lib/python3.8/site-packages/PyInstaller/loader/pyimod03_importers.py:489: in exec_module
    exec(bytecode, module.__dict__)
matplotlib/__init__.py:1015: in <module>
    ???
matplotlib/__init__.py:878: in rc_params
    ???
matplotlib/__init__.py:731: in matplotlib_fname
    ???
matplotlib/__init__.py:729: in gen_candidates
    ???
matplotlib/__init__.py:276: in wrapper
    ???
matplotlib/__init__.py:653: in _get_data_path
    ???
matplotlib/cbook/deprecation.py:111: in warn_deprecated
    ???
matplotlib/cbook/__init__.py:1993: in _warn_external
    ???
E   matplotlib.cbook.deprecation.MatplotlibDeprecationWarning:
E   The MATPLOTLIBDATA environment variable was deprecated in Matplotlib 3.1 and will be removed in 3.3.
ERROR: InvocationError for command /home/travis/build/astropy/astropy/.pyinstaller/run_astropy_tests (exited with code 4)
```

Naive re-pinning didn't work (#10779).

Also see: #10639

### PR DESCRIPTION
**Pull request by embray**
The warning is caused by PyInstaller's runtime hook for matplotlib, and prevents the Astropy tests from running without error when testing the PyInstaller build in our CI.
    
This workaround does not fix the issue in general: e.g. users using PyInstaller to bundle an application that users Astropy will have to include a similar fix in their own code (as the issue is not unique to Astropy).

EDIT: Fix #10785

### CODE REVIEW

**Comment by embray** [comment]
Confirmed that this fixes the pyinstaller job: https://travis-ci.org/github/astropy/astropy/jobs/732558269
I will drop afb1d95 from the PR now.

**Review by pllim — APPROVED** [review]
Thanks! Fine by me but I'll let @astrofrog have a look first before merge.

Also @astrofrog and @eteq can move the milestone if this didn't make it into 4.1 final.

**Review by astrofrog — APPROVED** [review]
Looks good to me, thanks for fixing this @embray!

File: astropy/conftest.py
Comment (astrojuanlu) [inline astropy/conftest.py:25]: For future reference, is there already an open issue in PyInstaller about this?

Comment (pllim) [reply astropy/conftest.py:25]: https://github.com/pyinstaller/pyinstaller/issues/3959#issuecomment-496878862 ?

Comment (embray) [reply astropy/conftest.py:25]: There is also a draft PR to fix it.  It's been delayed for a bit and still seems to have some kinks to work out, but I reviewed it myself and it seems sensible.  I thought of trying to integrate it into this fix, but it seems to be a bit tricky to override existing runtime hooks, so I decided it could get too complicated https://github.com/pyinstaller/pyinstaller/pull/5006

**Pull request merged**
